src/Build.UnitTests/WarningsAsMessagesAndErrors_Tests.cs

```diff
@@ -315,6 +315,116 @@ public void TaskReturnsHasLoggedErrorAndLogsWarningAsError_BuildShouldStopAndFai
             }
         }
 
+        [Fact]
+        public void TaskReturnsHasLoggedErrorAndLogsWarningAsError_WarningIsAlsoMessage_BuildShouldContinue()
+        {
+            using (TestEnvironment env = TestEnvironment.Create(_output))
```

Comment:
I don't normally look critically at tests, but please make sure your tests are substantially different from each other. I diffed your first and third tests with the second one, and they differed by one and two lines respectively. That could trivially become a theory and cut out almost a hundred lines of code.

Reply:
Fair point. Streamlined the tests, also realized the two batched builds test were covering the same scenario. Removed one of them.